Make reactive-oracle-client depend on jdbc-oracle

tsegismont opened the pull request:
Fixes #25415

Reactive Oracle Client extension does not work in native mode.
Instead of duplicating code, this PR makes the reactive client extension depend on jdbc client extension.

Also, it enables Reactive Oracle Client integration tests in native mode.

tsegismont committed:
Make reactive-oracle-client depend on jdbc-oracle

Fixes #25415

Reactive Oracle Client extension does not work in native mode.
Instead of duplicating code, this commit makes the reactive client extension depend on jdbc client extension.

tsegismont committed:
Enable Reactive Oracle Client integration tests in native mode

tsegismont commented:
cc @rsvoboda @Sanne

cescoffier commented on .github/native-tests.json:
```diff
@@ -45,7 +45,7 @@
         {
             "category": "Data7",
             "timeout": 65,
```
Should we increase it a bit?

tsegismont replied:
It does not seem necessary: `Quarkus CI / Native Tests - Data7 (pull_request) Successful in 25m`

cescoffier reviewed (approved):
LGTM, let's see what the CI says.

tsegismont commented:
The test failures looks unrelated:

```
[INFO] Running io.quarkus.platform.catalog.RegistrySnapshotCatalogCompatibilityTest
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.01 s <<< FAILURE! - in io.quarkus.platform.catalog.RegistrySnapshotCatalogCompatibilityTest
Error:  io.quarkus.platform.catalog.RegistrySnapshotCatalogCompatibilityTest.testRegistrySnapshotPlatformCatalog  Time elapsed: 0 s  <<< ERROR!
java.lang.IllegalStateException: Failed to locate the dev tools config file (.quarkus/config.yaml)
```

Sanne commented on extensions/reactive-oracle-client/runtime/pom.xml:
```diff
@@ -15,6 +15,14 @@
     <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jdbc-oracle</artifactId>
```
humm that seems wrong, no? it would pull in all transitive dependencies to actually start a JDBC pool

tsegismont replied:
Can you elaborate about the JDBC specific registrations?

Sanne reviewed (changes requested):
I don't think it's wise to pull in the JDBC quarkus extension as a dependency as that will trigger transitive dependencies which are necessary to setup a JDBC pool, and trigger other build items related to JDBC use of Oracle - which we likely want to skip in this case.

Could you extract a common module for the stuff you need to reuse? Or actually I'm not even sure if there's much in common, you most likely want to reuse *some* of the native-image support rules but they are most likely going to be needing different tuning.

Sanne reviewed (changes requested):
I meant to suggest making some changes.

tsegismont committed:
Fix failure to resolve implicit db kind

When there are multiple candidates for the same kind (i.e. reactive + jdbc), an implicit db kind can be chosen instead of returning an empty optional.

tsegismont commented:
@Sanne @cescoffier since we've come to an agreement about this PR, can we go ahead and merge it? The single CI failure is unrelated.

Sanne reviewed (approved):
Ok - sorry for the delay, have been travelling.
